app/assets/javascripts/utilities/buildCommentFormHTML.js.erb

```diff
@@ -19,31 +19,41 @@ function buildCommentFormHTML(commentableId, commentableType, parentId) {
-                <img alt="upload image" class="icon-image" src="<%= asset_path("image-upload.svg") %>" />\
-              </button>\
-              <label  class="image-upload-file-label" id="image-upload-file-label-' + randomIdNumber + '"></label>\
-              <input type="submit" id="image-upload-submit-' + randomIdNumber + '" value="Upload" style="display:none">\
-              <input class="uploaded-image" id="uploaded-image-' + randomIdNumber + '" />\
-            </div>\
-            <div class="actions reply-actions">\
-              <a href="#" class="cancel" onclick="cancel(event,this)">CANCEL</a>\
-              '+previewButton+'\
-              <input type="submit" class="comment-action-button" name="commit" value="SUBMIT" />\
+            <input class="uploaded-image" id="uploaded-image-' + randomIdNumber + '" />\
+            <div class="comment-bottom-actions">\
+              <div class="comment-editing-actions">\
+                <a href="/p/editor_guide" class="markdown-guide" target="_blank" title="Markdown Guide">\
+                  <img alt="markdown guide" class="icon-image" src="<%= asset_path('info.svg') %>" />\
+                </a>\
+                <div class="editor-image-upload">\
+                  <input type="file" id="image-upload-' + randomIdNumber + '"  name="file" accept="image/*" style="display:none">\
+                  <button title="Upload Image" class="image-upload-button" id="image-upload-button-' + randomIdNumber + '" onclick="handleImageUpload(event,'+ randomIdNumber + ')">\
+                    <img alt="upload image" class="icon-image" src="<%= asset_path("image-upload.svg") %>" />\
+                  </button>\
+                  <label  class="image-upload-file-label" id="image-upload-file-label-' + randomIdNumber + '"></label>\
+                  <input type="submit" id="image-upload-submit-' + randomIdNumber + '" value="Upload" style="display:none">\
+                </div>\
+                <div class="editor-response-templates">\
+                  <button class="response-templates-button" title="Use a response template" type="button">\
+                    <img src="<%= asset_path("response-template.svg") %>" alt="" class="icon-image">\
+                  </button>\
+                </div>\
+              </div>\
+              <div class="comment-submit-actions actions reply-actions">\
+                <a href="#" class="cancel" onclick="cancel(event,this)">CANCEL</a>\
+                '+previewButton+'\
+                <input type="submit" class="comment-action-button" name="commit" value="SUBMIT" />\
+              </div>\
             </div>\
           </form>';
 }
 
 function cancel(event, el) {
   event.preventDefault();
-  replaceActionButts(el.parentNode.parentNode.parentNode.parentNode);
+  replaceActionButts(el.parentNode.parentNode.parentNode.parentNode.parentNode);
```

Comment:
What is `el.parentNode.parentNode.parentNode.parentNode.parentNode` referencing? There may be a better way to get a reference to it. If not, consider creating a variable so it can be described and/or consider a comment explaining what it refers to.